tests/linearizability/watch.go

```diff
@@ -82,28 +82,37 @@ func validateWatchResponses(t *testing.T, responses [][]watchResponse, expectPro
 }
 
 func validateMemberWatchResponses(t *testing.T, responses []watchResponse, expectProgressNotify bool) {
-	var gotProgressNotify = false
-	var lastRevision int64 = 1
+	var (
+		gotProgressNotify       = false
+		lastEventRevision int64 = 1 // The event.Kv.ModRevision in the latest event.
+		lastHeadRevision  int64 = 1 // The resp.Header.Revision in last watch response.
+	)
 	for _, resp := range responses {
-		if resp.Header.Revision < lastRevision {
-			t.Errorf("Revision should never decrease")
+		if resp.Header.Revision < lastEventRevision || resp.Header.Revision < lastHeadRevision {
```

Comment:
Could you please split it into 2 ifs ? 
 - failure that event revisions are decreasing should be different from the response revisions.

Reply:
Since we print all events (lastHeader.Revision, lastEvent.ModRevision, current header.Revision), so we have enough info once running into this issue.

Updated anyway & thx.